File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int lowest_price = prices[0];
        int highest_profit = 0;
        for(int todays_price : prices){
            highest_profit = Math.max(highest_profit, todays_price - lowest_price);
            lowest_price = Math.min(lowest_price, todays_price);
        }
        return highest_profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int profit = 0;
        
        for(int i=0;i<prices.Length;i++){
            if(prices[0]<prices[i]){
                profit = Math.Max(profit, prices[i] - prices[0]);
            }else{
                prices[0] = prices[i];
            }
        }
        return profit;
    }
}